### PR TITLE
Fix crash when modifying preferences on non-isolated thread

### LINKED ISSUE #7930: Crash on FilterListResourceDownloader
### Description: 
We see reports of crashes on `FilterListResourceRownloader` when reading and writing to Preferences

### Steps to Reproduce 
Cannot reproduce as only stack traces are available

### Actual result: 
Crashes

### Expected result:
Should not crash

### Reproduces how often: [Easily reproduced, Intermittent Issue]
Intermittent Issue

### Brave Version: 
1.57

### Device details: 
N/A

### Website problems only:
- did you check with Brave Shields down? 
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information
Need to main actor isolate the Preferences read/write

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #7930

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Preferences/Preferences.swift [hunk 1]
```diff
@@ -79,6 +79,14 @@ extension Preferences {
...
       value = defaultValue
     }
+    
+    // An accessor to `value` which is MainActor isolated in case we need to ensure
+    // Note it would be best to make `value` MainActor isolated
+    // but this results in very large changes
+    @MainActor public var isolatedValue: ValueType {
+      get { return value }
+      set { value = newValue }
+    }
```
Comment (kylehickinson) [inline Sources/Preferences/Preferences.swift]: This doesn't make sense to add, either a type _requires_ actor isolation for a specific reason or it doesn't, you shouldn't add a new property thats isolated that sits side-by-side with a nonisolated version.

Need to reconsider _why_ you need actor isolation for a pref and fix that instead. If the whole thing is that all prefs need to be main actor isolated in general, then it should annotated on the class itself, not on a new property

**Review by kylehickinson — CHANGES_REQUESTED** [review]

**Comment by soner-yuksel** [comment]
What crash is being fixed btw?

**Comment by cuba** [comment]
@kylehickinson if you're ok with this I can also make an issue to make `Preferences` or `value` `MainActor` isolated.

File: Sources/Preferences/Preferences.swift (hunk 1 above)
Comment (cuba) [reply Sources/Preferences/Preferences.swift]: Ok I'll remove this, this isn't the fix it's just some compiler help. But I suggest we make an issue to isolate Preferences.

Comment (iccub) [reply Sources/Preferences/Preferences.swift]: @kylehickinson should we main actor isolate preferences in general?
Then I would even make it in separate commit

**Comment by kylehickinson** [comment]
Spoke in DM, proposed solution is to not involve Preferences at all for now, wrap the update in a `MainActor.run` as its not technically a problem with NSUserDefaults which is technically thread-safe.  We can still consider adding MainActor isolation to prefs in general in the future however

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**